src/pybind/mgr/rgw/module.py

```diff
@@ -311,18 +311,22 @@ def rgw_zone_create(self,
                         start_radosgw: Optional[bool] = True,
                         zone_endpoints: Optional[str] = None,
                         inbuf: Optional[str] = None) -> Any:
+
         if inbuf:
             try:
                 rgw_specs = self._parse_rgw_specs(inbuf)
             except RGWSpecParsingError as e:
                 return HandleCommandResult(retval=-errno.EINVAL, stderr=f'{e}')
         elif (zone_name and realm_token):
             token = RealmToken.from_base64_str(realm_token)
-            placement_spec = PlacementSpec.from_string(placement) if placement else None
+            if (type(placement) == dict):
+                placement_spec = PlacementSpec.from_json(placement) if placement else None
+            elif(type(placement) == str):
+                placement_spec = PlacementSpec.from_string(placement) if placement else None
```

Comment:
With the changes I proposed above there will be no need to add this check here. Placement can be passed as string and the typing in the method signature already expects a string.